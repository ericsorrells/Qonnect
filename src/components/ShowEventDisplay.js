// ========================================================================================
import React from 'react';
// ========================================================================================
import { formatTime, formatDateLong } from '../utils/utils';
// ========================================================================================

const ShowEventDisplay = (props) => {
  const { event } = props;
  return (
    <div className='modal__container'>
      <h1 className='event-display__title'>
        { event && event.eventName }
      </h1>
      <div className='event-display__name-location'>
        { event && event.date     && <EventInfo name='Date:'         info={formatDateLong(event.date)} />}
        { event && event.time     && <EventInfo name='Time:'         info={formatTime(event.time)}     />}
        { event && event.userName && <EventInfo name='Host/Hostess:' info={event.userName}             />}
        { event && event.location && <EventInfo name='Location:'     info={event.location}             />}
      </div>
      <div>
        {
          event && event.category &&
          <div className='event-display__description'>
            <div className='event-display__subtitle'>Description: </div>
            <div className='event-display__info'> {event.description} </div>
          </div>
        }
      </div>
    </div>
  )
}

const EventInfo = ({name, info}) => {
  return(
    <div>
      <div className='event-display__name-info'>
        <span className='event-display__name'>{name}</span>
        <span className='event-display__info'>{info}</span>
      </div>
    </div>
  )
}

export default ShowEventDisplay;
