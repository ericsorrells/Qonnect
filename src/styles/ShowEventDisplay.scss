.event-display__title {
  border-bottom: 1px solid $grey-text;
  margin-top: 0px;
}

.event-display__subtitle {
  margin: 0;
  font-weight: bold;
  margin-bottom: 10px;
}

.event-display__date-time {
  display: flex;
  color: $grey-text;
}

.event-display__name-location {
  margin: 10px 0;
}

.event-display__date-time-item {
  margin-right: 30px;
}

.event-display__name-info {
  margin-bottom: 10px;
}

.event-display__info {
  color: $grey-text;
}

.event-display__description {
  display: flex;
  flex-direction: column;
  margin-bottom: 10px;
}

.event-display__name {
  color: black;
  font-weight: bold;
  margin-right: 10px;
}