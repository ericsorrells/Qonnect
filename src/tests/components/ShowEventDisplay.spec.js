// ========================================================================================
import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import mockStoreSetup from '../../utils/mockStoreHelpers';
import 'jest-enzyme';
// ========================================================================================
import ShowEventDisplay from '../../components/ShowEventDisplay';
import { basicEvent } from '../stubs/event';
// ========================================================================================

describe('<ShowEventDisplay/>', () => {
  it('should render correctly', () => {
    const component = shallow(<ShowEventDisplay event={basicEvent} />)
    expect(component).toMatchSnapshot();
  });
});