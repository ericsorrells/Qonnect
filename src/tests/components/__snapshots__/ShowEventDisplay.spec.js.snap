// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowEventDisplay/> should render correctly 1`] = `
<div
  className="modal__container"
>
  <h1
    className="event-display__title"
  >
    Cool Event Title
  </h1>
  <div
    className="event-display__name-location"
  >
    <EventInfo
      info="Sunday January 0th 1970"
      name="Date:"
    />
    <EventInfo
      info="02:13 AM"
      name="Time:"
    />
    <EventInfo
      info="Rachel Deckard"
      name="Host/Hostess:"
    />
    <EventInfo
      info="123 Elm St."
      name="Location:"
    />
  </div>
  <div>
    <div
      className="event-display__description"
    >
      <div
        className="event-display__subtitle"
      >
        Description: 
      </div>
      <div
        className="event-display__info"
      >
         
        Lorem ipsum text
         
      </div>
    </div>
  </div>
</div>
`;
